ext/opcache/Optimizer/sccp.c

```diff
@@ -834,6 +834,8 @@ static inline int ct_eval_func_call(
 				|| zend_string_equals_literal(name, "rtrim")
 				|| zend_string_equals_literal(name, "ltrim")
 				|| zend_string_equals_literal(name, "str_split")
+				|| zend_string_equals_literal(name, "strtoupper")
+				|| zend_string_equals_literal(name, "strtolower")
```

Comment:
Unfortunately, these functions are locale dependent.
May be we may evaluate them for ASCII arguments.

Reply:
It's been a while and I forgot about strtoupper being locale-dependent.

strtoupper('i') is locale-dependent, and returns 'i' in Turkish instead of 'I', because the accented character it should be isn't ASCII.

Reply:
I don't think it's worth the trouble.